In [4]:
from sklearn.linear_model import LogisticRegression
import pandas as pd

# Cargar datos
trainX = pd.read_csv(r"C:\Users\Catalina\OneDrive\Escritorio\IA\Proyecto4-IA\Datos\TrainX.csv")
trainY = pd.read_csv(r"C:\Users\Catalina\OneDrive\Escritorio\IA\Proyecto4-IA\Datos\TrainY.csv").iloc[:,0]

validationX = pd.read_csv(r"C:\Users\Catalina\OneDrive\Escritorio\IA\Proyecto4-IA\Datos\ValidationX.csv")
validationY = pd.read_csv(r"C:\Users\Catalina\OneDrive\Escritorio\IA\Proyecto4-IA\Datos\ValidationY.csv").iloc[:,0]

testX = pd.read_csv(r"C:\Users\Catalina\OneDrive\Escritorio\IA\Proyecto4-IA\Datos\TestX.csv")
testY = pd.read_csv(r"C:\Users\Catalina\OneDrive\Escritorio\IA\Proyecto4-IA\Datos\TestY.csv").iloc[:,0]

# Entrenar el modelo
clf = LogisticRegression(random_state=0, max_iter=1000)
clf.fit(trainX, trainY)

# Seleccionar 5 datos muy diferentes (extremos del dataset)
samples = trainX.iloc[[0, len(trainX)//4, len(trainX)//2, len(trainX)//4*3, len(trainX)-1], :]

# Predicción de clases
print(clf.predict(samples))

# Probabilidades de cada clase
print(clf.predict_proba(samples))

# Exactitud en test
print(clf.score(testX, testY))


[1. 0. 1. 0. 1.]
[[1.10434632e-01 8.89565368e-01]
 [9.98890051e-01 1.10994947e-03]
 [4.81979374e-01 5.18020626e-01]
 [9.99806772e-01 1.93227530e-04]
 [2.00068190e-02 9.79993181e-01]]
0.9875


Resultados obtenidos: Se entrenó un modelo de Regresión Logística para predecir una variable binaria (0 o 1). Para evaluar su comportamiento, se seleccionaron cinco registros muy diferentes entre sí tomando los extremos del dataset: el primer registro, el último y otros ubicados aproximadamente en el 25%, 50% y 75% del conjunto de datos y se realizaron predicciones sobre ellos. El modelo generó tanto la clase predicha (0 o 1) como las probabilidades asociadas a cada clase, las cuales estuvieron siempre entre 0 y 1, indicando qué tan seguro estaba el modelo en cada decisión. Las predicciones mostraron probabilidades muy altas, lo que refleja confianza en los resultados obtenidos. Finalmente, al evaluar el modelo con los datos de prueba, se obtuvo un accuracy de 98.75%, evidenciando un excelente desempeño y una correcta capacidad de clasificación en este problema binario.